### PR TITLE
Implemented db migration feature

### PR DESCRIPTION
**Pull request by rkovalenko**

### CODE REVIEW

File: plugin/src/main/java/com/epam/grandhackathon/deployment/sphere/plugin/metadata/persistence/dao/DatabaseMigrator.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+import org.flywaydb.core.Flyway;
+import org.flywaydb.core.api.MigrationInfo;
+
+import com.google.common.io.ByteStreams;
+
+@Log
+public class DatabaseMigrator {
+	
+	private static final Character PATH_DELIMETER = '\\';
```
Comment (webdizz) [inline plugin/src/main/java/com/epam/grandhackathon/deployment/sphere/plugin/metadata/persistence/dao/DatabaseMigrator.java]: please use OS specific

File: plugin/src/main/java/com/epam/grandhackathon/deployment/sphere/plugin/metadata/persistence/dao/DatabaseMigrator.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	
+	public void migrate(@NonNull DataSource dataSource) {
+		log.info("Db migration started");
+		Flyway flyway = new Flyway();
+		flyway.setLocations(prepareMigrationLocation());
+		flyway.setBaselineOnMigrate(true);
+		flyway.setTable(DB_VERSION_TABLE);
+		flyway.setDataSource(dataSource);
+		flyway.setClassLoader(this.getClass().getClassLoader());
+		for (MigrationInfo i : flyway.info().all()) {
```
Comment (webdizz) [inline plugin/src/main/java/com/epam/grandhackathon/deployment/sphere/plugin/metadata/persistence/dao/DatabaseMigrator.java]: I believe it's not necessary FWD logs enough information

File: plugin/src/main/java/com/epam/grandhackathon/deployment/sphere/plugin/metadata/persistence/dao/DatabaseMigrator.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	}
+	
+	private String prepareMigrationLocation() {
+		try {
+			JarFile jarFile = getClassJarFile(DatabaseMigrator.class);
+			String jarLocation = jarFile.getName().replace(JAR_FILE_NAME, "");
+			copyMigrationToClassPath(jarFile, MIGRATION_SOURCE_PATH, jarLocation + MIGRATION_DIST_PATH);
+			return "filesystem:" + jarLocation + MIGRATION_DIST_PATH;
+		} catch (IOException e) {
+			log.warning("Loading migration files error: " + e.getMessage());
```
Comment (webdizz) [inline plugin/src/main/java/com/epam/grandhackathon/deployment/sphere/plugin/metadata/persistence/dao/DatabaseMigrator.java:49]: do not loose original exception

File: plugin/src/main/java/com/epam/grandhackathon/deployment/sphere/plugin/metadata/persistence/dao/DatabaseMigrator.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+		String path = CLASSPATH_DELIMETER + clazz.getName().replace('.', CLASSPATH_DELIMETER) + ".class";
+		String url = clazz.getResource(path).toString();
+		String jarUriPrefix = "jar:file:";
+		return new JarFile(url.substring(jarUriPrefix.length() + 1, url.indexOf("!")));
+	}
+
+	private void copyMigrationToClassPath(JarFile jarFile, String jarDirectory, String destinationDirectory) throws IOException {
+		for (Enumeration<JarEntry> entries = jarFile.entries(); entries.hasMoreElements();) {
+			JarEntry entry = entries.nextElement();
+			if (entry.getName().startsWith(jarDirectory + CLASSPATH_DELIMETER) && !entry.isDirectory()) {
```
Comment (webdizz) [inline plugin/src/main/java/com/epam/grandhackathon/deployment/sphere/plugin/metadata/persistence/dao/DatabaseMigrator.java]: looks quite complex
extract condition into dedicated method with proper name

File: plugin/src/main/resources/db/migration/V0.1.0__Init_Version.sql [hunk 5]
```diff
@@ -0,0 +1,22 @@
+CREATE TABLE IF NOT EXISTS BUILDS (
```
Comment (webdizz) [inline plugin/src/main/resources/db/migration/V0.1.0__Init_Version.sql:1]: IF NOT EXISTS is not necessary enymore

Comment (rkovalenko) [inline plugin/src/main/resources/db/migration/V0.1.0__Init_Version.sql:1]: Initial migration will be executed on non-empty db schema. 
I think it should be done after initial migration

Comment (webdizz) [inline plugin/src/main/resources/db/migration/V0.1.0__Init_Version.sql:1]: it's ok, imagine we're just starting

**Pull request merged**